### PR TITLE
feat(amplify-frontend-ios): amplify-xcode integration

### PR DESCRIPTION
**Pull request by diegocstn**
amplify-xcode integration, JS bindings code generation

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by edwardfoyle — COMMENTED** [review]
Left a few comments. But in general I'm a little confused by the approach of auto-generating the amplify-xcode.js file. It seems like an unnecessary layer of indirection and complexity. Could the relevant logic from native-bindings-codegen/index.js that parses the amplify-xcode schema be lifted into functions that are called directly by the event handlers?

File: packages/amplify-frontend-ios/lib/amplify-xcode.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+/**
+ * Import Amplify configuration files
+ * @param {Object} params
+ * @param {String} params.path Project base path
+ */
+async function importConfig(params) {
+  let command = `${amplifyXcodePath} import-config`;
+  if (params['path']) {
+    command += ` --path=${params['path']}`;
```
[the remark this answers is not in the document]
Comment (diegocstn) [reply packages/amplify-frontend-ios/lib/amplify-xcode.js:36]: options can have dashes in their name (i.e. `output-path`), so it's safer to always use bracket notation

File: packages/amplify-frontend-ios/native-bindings-codegen/index.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
+const path = require('path');
+const fs = require('fs');
+const _ = require('lodash');
+
+const CODEGEN_TEMPLATES_FOLDER = 'templates';
+
+const canonicalFunctionName = commanName => _.camelCase(commanName);
```
Comment (edwardfoyle) [inline packages/amplify-frontend-ios/native-bindings-codegen/index.js:7]: typo: commonName

File: packages/amplify-frontend-ios/lib/amplify-xcode.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+// http://aws.amazon.com/apache2.0
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+//
+
+// ############################################################################
+// Auto-generated file using `build` NPM script. Do not edit this file manually
```
[the remark this answers is not in the document]
Comment (diegocstn) [reply packages/amplify-frontend-ios/lib/amplify-xcode.js:17]: discussed offline about code generation, moved to a different NPM script

File: packages/amplify-frontend-ios/package.json [hunk 4]
```diff
@@ -16,13 +16,15 @@
     "aws"
   ],
   "scripts": {
+    "generate-xcode-bindings": "node ./scripts/native-bindings-gen",
     "test": "jest",
     "test-watch": "jest --watch"
   },
   "dependencies": {
-    "amplify-app": "2.23.3",
```
Comment (diegocstn) [inline packages/amplify-frontend-ios/package.json:23]: removed `amplify-app` dependency as the package will rely on `amplify-xcode`

File: packages/amplify-frontend-ios/index.js [hunk 5]
```diff
@@ -59,22 +59,35 @@ async function executeAmplifyCommand(context) {
...
-      platform: frontend,
-      skipInit: true,
-      internalOnlyIosCallback: true,
-    });
+  const isMacOs = process.platform === 'darwin';
+  if (!isFrontendiOS || !isXcodeIntegrationEnabled || !isMacOs) {
+    return;
+  }
+  context.print.info('Updating iOS project');
+  const projectPath = process.cwd();
```
Comment (edwardfoyle) [inline packages/amplify-frontend-ios/index.js:71]: does this break if the command is run from a subdirectory of the project?

**Review by edwardfoyle — APPROVED** [review]
just left one clarification question

File: packages/amplify-frontend-ios/index.js (hunk 5 above)
Comment (diegocstn) [reply packages/amplify-frontend-ios/index.js:71]: Thanks for pointing that out, that's an `amplify-app` relic. It doesn't break but throws an error. 
I'll replace with `pathManager.findProjectRoot()` from `amplify-cli-core`

**Review by edwardfoyle — APPROVED** [review]

**Comment by edwardfoyle** [comment]
I think you need to merge / rebase master to get the tests to pass

**Pull request merged**